File: nodes/workflows/Workflows.Services/Query/PackageQuery.cs
using Distributed.Core.Services;
using Microsoft.EntityFrameworkCore;
using Workflows.Contracts;
using Workflows.Data;

namespace Workflows.Services;
public class PackageQuery : EntityQuery<Package, WorkflowsContext>
{
    public PackageQuery(WorkflowsContext db) : base(db)
    { }

    public async Task<List<Package>> GetByType(string entityType) =>
        await Set
            .Where(x =>
                x.EntityType == entityType
            )
            .ToListAsync();

    public async Task<List<Package>> GetByEntity(int id, string entityType) =>
        await Set
            .Where(x =>
                x.EntityId == id
                && x.EntityType == entityType
            )
            .ToListAsync();

    public async Task<Package?> GetActivePackage(int id, string entityType) =>
        await Set
            .FirstOrDefaultAsync(x =>
                x.EntityId == id
                && x.EntityType == entityType
                && (
                    x.State == PackageStates.Pending
                    || x.State == PackageStates.Returned
                )
            );
}

File: apps/libs/distributed/projects/toolkit/src/nodes/package-query.ts
import {
    Inject,
    Injectable
} from '@angular/core';

import { HttpClient } from '@angular/common/http';
import { EntityQuery } from '@distributed/core';
import { Package } from '@workflows/contracts';
import { firstValueFrom } from 'rxjs';
import { SnackerService } from '../services';

@Injectable()
export class PackageQuery extends EntityQuery<Package> {
    constructor(
        @Inject('packageApiUrl') endpoint: string,
        http: HttpClient,
        snacker: SnackerService
    ) {
        super(endpoint, http, snacker);
    }

    protected getByType$ = (entityType: string) =>
        this.http.get<Package[]>(`${this.api}getByType/${entityType}`);

    protected getByEntity$ = (id: number, entityType: string) =>
        this.http.get<Package[]>(`${this.api}getByEntity/${id}/${entityType}`);

    protected getActivePackage$ = (id: number, entityType: string) =>
        this.http.get<Package>(`${this.api}getActivePackage/${id}/${entityType}`);

    getByType = (entityType: string) =>
        firstValueFrom(this.getByType$(entityType));

    getByEntity = (id: number, entityType: string) =>
        firstValueFrom(
            this.getByEntity$(id, entityType)
        );

    getActivePackage = (id: number, entityType: string) =>
        firstValueFrom(
            this.getActivePackage$(id, entityType)
        );
}